src/controllers/comment.js

```diff
@@ -0,0 +1,110 @@
+import models from '../db/models';
+import { errorResponse, commentResponse, getCommentsAuthors } from '../helpers';
+
+const { Comment: CommentModel } = models;
+
+/**
+ * @description A controller class for handling comment business logic
+ *
+ * @class Comment
+ */
+class Comment {
+  /**
+   * @description - Gets all comments associated with an article
+   *
+   * @static
+   * @param {object} request - Request sent to the router
+   * @param {object} response - Response sent from the controller
+   *
+   * @returns {object} - containing an array of comments
+   *
+   * @memberof Comment
+   */
+  static async getAll(request, response) {
+    const { article } = request;
+    try {
+      const comments = await article.getComments({
+        attributes: ['id', 'createdAt', 'userId', 'updatedAt', 'body']
+      });
+      const authors = await getCommentsAuthors(comments);
+      response.send(commentResponse(comments, authors));
+    } catch (error) {
+      response.status(500).send(errorResponse([error.message]));
+    }
+  }
+
+  /**
+   * @description - Gets an article comment by it's ID
+   *
+   * @static
+   * @param {object} request - Request sent to the router
+   * @param {object} response - Response sent from the controller
+   *
+   * @returns {object} - a single comment
+   *
+   * @memberof Comment
+   */
+  static async getOne(request, response) {
+    const { article, params } = request;
+    const { commentId } = params;
+
+    try {
+      const comments = await article.getComments({
+        attributes: ['id', 'createdAt', 'userId', 'updatedAt', 'body'],
+        where: { id: parseInt(commentId, 10) }
+      });
+
+      if (!comments.length) return response.status(404).send(errorResponse([`Comment with id ${commentId} not found`]));
+
+      const authors = await getCommentsAuthors(comments);
+      const comment = commentResponse(comments, authors).comments[0];
+      response.send({ comment });
+    } catch (error) {
+      response.status(500).send(errorResponse(['Server error. Failed to get comment']));
+    }
+  }
+
+  /**
+   * @description - Create/Post a comment on an article
+   *
+   * @static
+   * @param {object} request - Request sent to the router
+   * @param {object} response - Response sent from the controller
+   *
+   * @returns {object} - object representing response message
+   *
+   * @memberof Comment
+   */
+  static async create(request, response) {
+    const { body: commentBody } = request.body;
+
+    if (!commentBody) return response.status(400).send(errorResponse(['Comment body is required']));
```

Comment:
@codeshifu I see you're validating here, what are your thoughts on abstracting the validation?

Reply:
This is not a repetitive task, it's just making sure something is in the comment. I would love to know why you feel this should be abstracted

Reply:
I was just asking because I felt that the controller should just be concerned about it's task, but I get your point though👍🏿